archivist/runner.py

```diff
@@ -82,113 +109,101 @@ def run_steps(self, config: Dict):
         if data:
             args.append(data)
-        LOGGER.debug("args %s", args)
 
-        if "confirm" in step:
-            kwargs["confirm"] = step["confirm"]
+        # wait for a number of seconds
+        wait_time = s.get("wait_time", 0)
+        if wait_time > 0:
+            LOGGER.info("Waiting for %d seconds", wait_time)
+            time_sleep(wait_time)
+
+        LOGGER.debug("args %s", args)
+        if len(kwargs) > 0:
             LOGGER.debug("kwargs %s", kwargs)
             response = action(*args, **kwargs)
         else:
             response = action(*args)
 
+        # Some actions return a tuple
+        if isinstance(response, tuple):
+            response = response[0]
+
         print_response = s.get("print_response")
         if print_response:
-            LOGGER.info("Response %s", json_dumps(response, indent=4))
-
-        self.set_entities(action_name, response)
+            if isinstance(response, GeneratorType):
+                for e in response:
+                    LOGGER.info("Response %s", json_dumps(e, indent=4))
+            else:
+                LOGGER.info("Response %s", json_dumps(response, indent=4))
 
-        if action_name == "COMPLIANCE_COMPLIANT_AT" and not response["compliant"]:
-            self.process_compliant_at(response)
+        self.set_entities(response)
 
-    def set_entities(self, action_name: str, response: Dict):
+    def set_entities(self, response: Dict):
         """sets entties entry"""
 
         try:
             name = response.name
         except AttributeError:
             pass
         else:
-            if name is not None:
-                self.entities[action_name][name] = response
+            self.entities[name] = response
 
-    def process_compliant_at(self, response: Dict):
-        """Evaluates Gets asset id"""
+    def asset_id(self, name: str) -> Optional[str]:
+        """Gets entity id"""
 
-        # if there is a reason print it
-        for outcome in response["compliance"]:
-
-            if outcome["compliant"]:
-                continue
-
-            # get the compliance policy
-            policy = self._archivist.compliance_policies.read(
-                outcome["compliance_policy_identity"]
-            )
-
-            # print the policy name and the reason
-            LOGGER.info(
-                "NON-COMPLIANCE -> Policy: %s: Reason %s",
-                policy["display_name"],
-                outcome["reason"],
-            )
-
-    def asset_id(self, asset_name: str) -> Optional[str]:
-        """Gets asset id"""
-
-        if asset_name is not None:
-            asset_id = self.entities["ASSETS_CREATE"][asset_name]["identity"]
-            if isinstance(asset_id, str):
-                return asset_id
+        asset_id = self.entities[name]["identity"]
+        if isinstance(asset_id, str):
+            return asset_id
 
         return None
-
-    def delete(self):
```

Comment:
Have we lost the delete function or has it gone somewhere else?

Reply:
it is now controlled by an explicit setting delete: true in the yaml file and I can see that I have forgotten to add the delete logic

Reply:
added the required logic and tests